### PR TITLE
Fix Privacy Center configuration and make consent card configurable

### LINKED ISSUE #1597: Privacy Center custom configuration fails without `config/types.ts`
### Bug Description

To customize the Fides Privacy Center, users can either copy the contents of the `clients/privacy-center` folder, or just use our published Docker image ([`ethyca/fides-privacy-center`](https://hub.docker.com/r/ethyca/fides-privacy-center)) and mount their own custom `/content` folder to replace the default config.json and config.css. Docs for this are here: https://ethyca.github.io/fides/dev/deployment/#set-up-the-privacy-center-optional

However, in a recent change we added some source files to that `/config` folder that now prevent the published image from rebuilding itself as needed unless you supply the exact `config/types.js` file from the codebase along with your own customer `config.json`, etc.

Now, while it's debatable whether we really _should_ be rebuilding (we should be able to load the config at runtime instead), or if we've missed a much better way to inject custom config into the image, but right now this is our preferred solution and it's not working well!

#### Steps to Reproduce

1. Build the most recent privacy center `dev` image
2. `docker run --rm -v $(pwd)/config:/app/config -p 3000:3000 ethyca/fides-privacy-center:dev`
3. Open localhost:3000

### Expected behavior

You should be able to customize the config.json fully without issue.

### Additional context

This'd be way easier to test if we included a custom config option in `nox -s test_env` to simulate this.

### PR DESCRIPTION
**Pull request by NevilleS**
Closes #1597

### Code Changes

* [X] Move types out of `/config` and into `/types`
* [X] Update privacy center tests
* [X] Mount a custom `/config` folder in docker for testing 
* [X] Add new configuration options for the consent card `logo_path`, `title`, and `description`

### Steps to Confirm

* [X] Run `nox -s test_env` and confirm that the build succeeds with a custom config & css for test environment
* [X] Run `nox -s dev` and `cd clients/privacy-center && npm run dev` and confirm that development build uses the default config from the source code

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation Updated:
  * [X] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
* [X] Issue Requirements are Met
* [X] Relevant Follow-Up Issues Created
* [X] Update `CHANGELOG.md`

### Description Of Changes

This started as a minor change to fix an issue that occurs in production (see #1597) where we've broken the ability to mount a custom `/config` folder to a running container. In testing further, I realized that the "Manage my consent" card was hardcoded which prevented the ability to style it, too. So this PR expanded a bit to include a new feature.

### CODE REVIEW

File: clients/privacy-center/components/Card.tsx [hunk 1]
```diff
@@ -28,11 +30,11 @@ const Card: React.FC<CardProps> = ({
     userSelect="none"
     m={2}
     _hover={{
-      boxShadow: "complimentary-2xl",
```
Comment (NevilleS) [inline clients/privacy-center/components/Card.tsx:31]: This shadow provided a purplish-trim by default, which must have been from old privacy center styling. It makes providing custom colours a real chore, so I've swapped this out with a nice box shadow from a recent Figma.

File: clients/privacy-center/components/Card.tsx [hunk 2]
```diff
@@ -20,6 +20,8 @@ const Card: React.FC<CardProps> = ({
     bg="white"
     py={8}
     px={6}
+    border="1px solid"
+    borderColor="gray.200"
```
Comment (NevilleS) [inline clients/privacy-center/components/Card.tsx:24]: Recent figma designs all have a small border around the card at all times for some contrast against the light background.

File: clients/privacy-center/config/config.css [hunk 3]
```diff
@@ -1 +1,19 @@
...
+    /* Header & highlight color */
+    /* --chakra-colors-gray-100: #EDF2F7; */
+    /* Modal text color */
+    /* --chakra-colors-gray-500: #718096; */
+    /* Body text color */
+    /* --chakra-colors-gray-600: #4A5568; */
+    /* Primary button hover color */
+    /* --chakra-colors-primary-400: #464B83; */
+    /* Primary button color */
+    /* --chakra-colors-primary-800: #111439; */
```
Comment (NevilleS) [inline clients/privacy-center/config/config.css:18]: I went through the exercise of theme-ing a privacy center and this was really painful to hunt down these theme colours... this still isn't ideal, but it provides some hints and living documentation for users in the future...

File: clients/privacy-center/config/config.json [hunk 4]
```diff
@@ -30,31 +30,34 @@
   ],
   "includeConsent": true,
   "consent": {
+    "icon_path": "/consent.svg",
+    "title": "Manage your consent",
+    "description": "Manage your consent preferences, including the option to select 'Do Not Sell My Personal Information'.",
```
Comment (NevilleS) [inline clients/privacy-center/config/config.json:35]: Using these new config options.

File: clients/privacy-center/config/config.json [hunk 5]
```diff
@@ -30,31 +30,34 @@
...
     "cookieName": "fides_consent",
     "consentOptions": [
       {
-        "fidesDataUseKey": "provide.service",
-        "name": "Provide our Service",
+        "fidesDataUseKey": "advertising",
+        "name": "Data Sales or Sharing",
+        "description": "We may use some of your personal information for behavioral advertising purposes, which may be interpreted as 'Data Sales' or 'Data Sharing' under regulations such as CCPA, CPRA, VCDPA, etc.",
+        "url": "https://example.com/privacy#data-sales",
```
Comment (NevilleS) [inline clients/privacy-center/config/config.json:42]: Just trying to continue to make the defaults as realistic as possible, for our own benefit

File: clients/privacy-center/pages/index.tsx [hunk 6]
```diff
@@ -108,6 +116,7 @@ const Home: NextPage = () => {
         >
           <Image
             src={config.logo_path}
+            margin="8px"
```
Comment (NevilleS) [inline clients/privacy-center/pages/index.tsx:119]: Logo was *very* tightly squeezed

File: docker-compose.yml [hunk 7]
```diff
@@ -62,7 +62,7 @@ services:
       - "3001:3000"
     volumes:
       - type: bind
-        source: ./clients/privacy-center/config
+        source: ./data/config/privacy_center_config
```
Comment (NevilleS) [inline docker-compose.yml:65]: Use the custom privacy center config when running the privacy center via docker. This forces us to ensure we don't break this volume mount pattern!

**Review by NevilleS — COMMENTED** [review]
Some inline comments

**Review by Kelsey-Ethyca — APPROVED** [review]

**Review by chriscalhoun1974 — APPROVED** [review]
lgtm

**Pull request merged**